src/components/Delegate/StakingOpportunities.tsx

```diff
@@ -73,10 +74,82 @@ export const ValidatorName = ({ moniker, isStaking }: ValidatorNameProps) => {
   )
 }
 
-export const StakingOpportunities = ({ assetId, opportunities }: StakingOpportunitiesProps) => {
-  const isLoaded = true
-  const isStaking = opportunities.some(x => x.cryptoAmount)
-  const assetSymbol = useAppSelector(state => selectAssetByCAIP19(state, assetId)).symbol
+export const StakingOpportunities = ({ assetId }: StakingOpportunitiesProps) => {
+  const stakingDataStatus = useAppSelector(selectStakingDataStatus)
+  const isStakingDataLoaded = stakingDataStatus === 'loaded'
+  const validatorStatus = useAppSelector(selectValidatorStatus)
+  const isValidatorDataLoaded = validatorStatus === 'loaded'
+  const isLoaded = isStakingDataLoaded && isValidatorDataLoaded
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+  const dispatch = useAppDispatch()
+
+  const accountSpecifiersForChainId = useAppSelector(state =>
+    selectPubkeyishByChainId(state, asset?.caip2)
+  )
+  const accountSpecifier = accountSpecifiersForChainId?.[0]
+
+  const stakingOpportunities = useAppSelector(state =>
+    selectStakingOpportunityDataByDenom(
+      state,
+      accountSpecifier,
+      SHAPESHIFT_VALIDATOR_ADDRESS,
+      ASSET_ID_TO_DENOM[asset.caip19]
+    )
+  )
+  const shapeshiftValidator = useAppSelector(state =>
+    selectSingleValidator(state, accountSpecifier, SHAPESHIFT_VALIDATOR_ADDRESS)
+  )
+  const stakingOpportunityDefault = [
+    {
+      ...shapeshiftValidator
```

Comment:
[gomesalexandre](https://github.com/gomesalexandre) [7 hours ago](https://github.com/shapeshift/web/pull/1356/files#r840737306)
Spread not needed here, you can pass the reference directly.

Reply:
IMO it also doesnt hurt to spread it to be safe and make clear it cant mutate the reference for peace of mind (even if it doesnt matter in this case)